### ISSUE TITLE
Domeneshop.dns.createRecord(domainId, record) - allows duplicate entries

### ISSUE THREAD

**Issue by Juksefantomet** [domeneshop/domeneshop.js#15]
[Domeneshop.dns.createRecord(domainId, record)](https://github.com/domeneshop/domeneshop.js#domeneshopdnscreaterecorddomainid-record)

When creating a record and said host already exists with the same type even on unique data and duplicate data it allows the create to proceed.

This is not possible via the GUI of domeneshop.

**Comment by eirikrye** [comment]
What type of data do you mean? For example, a duplicate A-record with same data/IP-address and same hostname is allowed, both in DNS and in our web-based control panel.

The only record type that *cannot* be duplicated is a CNAME record, as it's the only record that can exist for a hostname.

**Comment by Juksefantomet** [comment]
The issue was relevant 8 months ago when it was reported.

**Comment by eirikrye** [comment]
That's fair, but the issue you're experiencing would likely still be here as the API has not changed significantly in the past eight months. Apologies for answering this issue so late.

**Comment by Juksefantomet** [comment]
Hello, In short it's hard for me to reproduce the incident which occurred 8 months ago as I did not keep the snippet in hand which caused the issue in the first place. (My fault for not properly including that in the OP, apologies).

I have not been able to reproduce the incident with CNAME and can confirm the async creation is blocked when attempting to do so. The person who looked into this and reported the errors in the first place is not at work and available to confirm any "odd results" at this time.

The incident occurred when i was looping through existing domains and by accident inserted my createDns function within the subdomain looping which caused the created to trigger pr. subdomain iteration instead of a single time.

The issue was related to an async create request which resulted in me contacting the ops team with GUI access to the interface. I reported a weird and odd behavior around the subdomains in question and his response was.

"I have never in my life seen an entry record list like this. As it's not even possible to insert the subdomains as multiple records overlapping". Due to requirements i quickly deleted the dns records and simply recreated everything with the correct syntax to correct any co-related issues caused.

At the time i am 100% percent sure i was inserting subdomains for A records + IP and some with CNAME for a dns.

I shortly after this migrated away from this package at a later point when i discovered the API and created my own in ruby.
